system-tests/helpers/kafkahelpers.py

```diff
@@ -30,15 +28,15 @@ def publish_run_start_message(
     with open(nexus_structure_filepath, "r") as nexus_structure_file:
         nexus_structure = nexus_structure_file.read().replace("\n", "")
 
-    runstart_message = create_runstart_message(
+    runstart_message = serialise_pl72(
         job_id,
         nexus_filename,
         start_time,
         stop_time,
         nexus_structure=nexus_structure,
         service_id=service_id,
     )
-    producer.produce(topic, runstart_message)
+    producer.produce(topic, bytes(runstart_message))
```

Comment:
It seems to me that either there is a bug in `serialise_pl72` or there is a bug in the Flatbuffers library if use of `bytes` here is required. If it is not, it should be removed, otherwise the bug should be fixed in `serialise_pl72`.

Reply:
NICOS uses a different Kafka library which requires `bytearray` rather than `bytes`. So we have to cast here or in NICOS whichever type we choose to output from the library.